java/src/main/java/ai/rapids/cudf/HostMemoryBuffer.java

```diff
@@ -94,6 +117,23 @@ private HostMemoryBuffer(long address, long lengthInBytes, HostMemoryBuffer pare
     super(address, lengthInBytes, parent);
   }
 
+  /**
+   * Return a ByteBuffer that provides access to the underlying memory.  Please note <ol></ol> if the buffer is
+   * larger than a ByteBuffer can handle (2GB) only the first part of it will be exposed.  Also be
+   * aware that the ByteBuffer will be in native endian order, which is different from
+   */
+  public final ByteBuffer asByteBuffer() {
+    long len = Math.min(Integer.MAX_VALUE, length);
```

Comment:
Is there a use-case where silently truncating the buffer is appropriate?  I'm wondering if this should be turned into an exception instead to avoid silent data dropping, or we only expose the version with explicit offset/lengths.